x-pack/plugins/apm/public/components/alerting/rule_types/error_count_rule_type/index.tsx

```diff
@@ -125,7 +125,12 @@ export function ErrorCountRuleType(props: Props) {
     />,
     <EnvironmentField
       currentValue={params.environment}
-      onChange={(value) => setRuleParams('environment', value)}
+      onChange={(value) =>
+        setRuleParams(
+          'environment',
+          value !== '' ? value : ENVIRONMENT_ALL.value
```

Comment:
An alternative approach: making `environment` optional on the server-side, and thus letting the server-side handle it. This will make the API easier to consume for customers.
Not a blocker for this fix though.